analyses/hotspots-detection/00-subset-maf.R

```diff
@@ -0,0 +1,106 @@
+# K. S. Gaonkar 2021
+# In this script, we will filter maf from a caller into
+# a resulting filtered file that has all calls that
+# - overlap with amino acid positions in a curated and
+# published cancer hotspot [database](https://www.cancerhotspots.org/files/hotspots_v2.xls)
+# - overlap with non-coding region hotspots mutations ,
+
+
+suppressPackageStartupMessages(library("optparse"))
+suppressPackageStartupMessages(library("tidyverse"))
+
+option_list <- list(
+  make_option(c("-m", "--maffile"),
+    type = "character",
+    help = "Filtered calls from [mutect2|strelka2|vardict|lancet]"
+  ),
+  make_option(c("-c", "--caller"),
+    type = "character",
+    help = "Caller type [mutect2|strelka2|vardict|lancet]"
+  ),
+  make_option(c("-f", "--cancer_hotspot_folder"),
+    help = "MSKCC hotspot file", type = "character"
+  ),
+  make_option(c("-g", "--genomic_site_hotspot_file"),
+    help = "genomic hotspot location file", type = "character"
+  ),
+  make_option(c("-o", "--output_file"),
+    help = "file name for filtered calls ", type = "character"
+  )
+)
+
+# Get command line options, if help option encountered print help and exit,
+opt <- parse_args(OptionParser(option_list = option_list))
+caller <- opt$caller
+# Converting caller ID to call Uppercase for error handling
+caller <- tolower(caller)
+output_file <- opt$output_file
+
+maf_coltypes <- readRDS(file.path("input", "maf_coltypes.RDS"))
```

Comment:
I think she meant "2 consecutive bp change"

Reply:
I though of adding it as RDS mainly because the read_tsv needs readr::col_[data_types]() specifications. Probably there is a way to have a csv that I don't know yet.